File: java-undertow/src/test/java/org/dummy/calc/AppTest.java
package org.dummy.calc;

import java.util.concurrent.TimeUnit;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;

import io.undertow.Undertow;

public class AppTest extends AppTestBase {

    static Undertow SERVER = null;

    @BeforeAll
    public static void setUp() throws Exception {
        TimeUnit.SECONDS.sleep(1L);
        SERVER = App.undertow();
        SERVER.start();
    }

    @AfterAll
    public static void tearDown() {
        if (SERVER != null) {
            SERVER.stop();
        }
    }
}

File: java-ktor/src/test/kotlin/AppTest.kt
package org.dummy.calc

import io.ktor.server.netty.*
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit

class AppTest : AppTestBase() {

    companion object {
        private val serverExecutor: ExecutorService = Executors.newSingleThreadExecutor()

        @JvmStatic
        @BeforeAll
        fun startServer() {
            serverExecutor.submit {
                EngineMain.main(arrayOf("src/main/resources/application.yaml"))
            }
            TimeUnit.SECONDS.sleep(1L)
        }

        @JvmStatic
        @AfterAll
        fun stopServer() {
            serverExecutor.shutdown()
        }
    }
}
